main.go

```diff
@@ -0,0 +1,357 @@
+	// #nosec G304
+	nodeImageData, err := os.ReadFile(configFilePath)
+	if err != nil {
+		return fmt.Errorf("failed to read config.yaml: %w", err)
+	}
+
+	// Unmarshal YAML data into NodeImages struct
+	var nodeImages NodeImages
+	if err := yaml.Unmarshal(nodeImageData, &nodeImages); err != nil {
+		return fmt.Errorf("failed to unmarshal YAML: %w", err)
+	}
+
+	// Check if the URL already exists for the given image
+	var imageURLExists bool
+	for _, image := range nodeImages.OpenStackNodeImages {
+		if image.URL != "" {
+			imageURLExists = true
+			break
+		}
+	}
+	// If the URL doesn't exist, update it for the image
+	if !imageURLExists {
+		// Load registry configuration from YAML file
+		// #nosec G304
+		registryConfigFile, err := os.Open(registryConfigPath)
+		if err != nil {
+			return fmt.Errorf("error opening registry config file: %w", err)
+		}
+		defer registryConfigFile.Close()
+
+		var registryConfig RegistryConfig
+		decoder := yaml.NewDecoder(registryConfigFile)
+		if err := decoder.Decode(&registryConfig); err != nil {
+			return fmt.Errorf("error decoding registry config file: %w", err)
+		}
+		// Generate URL
+		newURL := fmt.Sprintf("%s%s/%s/%s", "https://", registryConfig.Config.Endpoint, registryConfig.Config.Bucket, imageName)
+		for i := range nodeImages.OpenStackNodeImages {
+			nodeImages.OpenStackNodeImages[i].URL = newURL
+			break
+		}
+
+		// Marshal the updated struct back to YAML
+		updatedNodeImageData, err := yaml.Marshal(&nodeImages)
+		if err != nil {
+			return fmt.Errorf("failed to marshal YAML: %w", err)
+		}
+
+		// Write the updated YAML data back to the file
+		if err := os.WriteFile(configFilePath, updatedNodeImageData, os.FileMode(0o644)); err != nil {
+			return fmt.Errorf("failed to write config.yaml: %w", err)
+		}
+
+		fmt.Printf("URL updated for image: %s\n", newURL)
+	} else {
+		fmt.Printf("URL already exists for the image\n")
+	}
+	return nil
+}
+
+func copyFile(src, dest string) error {
+	// #nosec G304
+	data, err := os.ReadFile(src)
+	if err != nil {
+		return fmt.Errorf("error reading source file: %w", err)
+	}
+
+	if err := os.WriteFile(dest, data, os.FileMode(0o644)); err != nil {
+		return fmt.Errorf("error writing to destination file: %w", err)
+	}
+
+	return nil
+}
+
+// GetConfig returns Config.
+func GetConfig(configPath string) (NodeImages, error) {
+	configFileData, err := os.ReadFile(filepath.Clean(configPath))
+	if err != nil {
+		return NodeImages{}, fmt.Errorf("failed to read config file: %w", err)
+	}
+
+	nd := NodeImages{}
+	if err := yaml.Unmarshal(configFileData, &nd); err != nil {
+		return NodeImages{}, fmt.Errorf("failed to unmarshal config yaml: %w", err)
+	}
+
+	if nd.APIVersion == "" {
+		return NodeImages{}, fmt.Errorf("api version must not be empty")
+	}
+
+	if len(nd.OpenStackNodeImages) == 0 {
+		return NodeImages{}, fmt.Errorf("at least one node image needs to exist in OpenStackNodeImages list")
+	}
+
+	// Ensure all fields in OpenStackNodeImages are defined
```

Comment:
IMO this check can be done more "inteligently" when createops will not be harcoded